include/nvexec/stream/algorithm_base.cuh

```diff
@@ -0,0 +1,144 @@
+
+            static constexpr ::std::size_t value = //
+                stdexec::__v< stdexec::__gather_completions_for<
+                stdexec::set_value_t,
+                Sender,
+                stdexec::env_of_t<Receiver>,
+                stdexec::__q<result_size_for_t>,
+                stdexec::__q<max_in_pack>>>;
+            };
+
+            operation_state_base_t<ReceiverId>& op_state_;
+            STDEXEC_NO_UNIQUE_ADDRESS Fun fun_;
+
+            public:
+            using __id = receiver_t;
+
+            constexpr static ::std::size_t memory_allocation_size = max_result_size::value;
+
+            template <stdexec::same_as<stdexec::set_value_t> _Tag, class Range>
+            friend void tag_invoke(_Tag, __t&& self, Range&& range) noexcept {
+                DerivedReceiver::set_value_impl((__t&&) self, (Range&&) range);
+            }
+
+            template <stdexec::__one_of<stdexec::set_error_t, stdexec::set_stopped_t> Tag, class... As>
+            friend void tag_invoke(Tag, __t&& self, As&&... as) noexcept {
+            self.op_state_.propagate_completion_signal(Tag(), (As&&) as...);
+            }
+
+            friend stdexec::env_of_t<Receiver> tag_invoke(stdexec::get_env_t, const __t& self) {
+            return stdexec::get_env(self.op_state_.receiver_);
+            }
+
+            __t(Fun fun, operation_state_base_t<ReceiverId>& op_state) : op_state_(op_state) , fun_((Fun&&) fun) {}
+        };
+    };
+
+    template <class SenderId, class Fun, class DerivedSender>
+    struct sender_t {
+        struct __t : stream_sender_base {
+            using Sender = stdexec::__t<SenderId>;
+            using __id = sender_t;
+
+            template<class Receiver>
+            using receiver_t = typename DerivedSender::template receiver_t<Receiver>;
+
+            template <class Range>
+            using set_value_t = typename DerivedSender::template set_value_t<Range>;
+
+            Sender sndr_;
+            STDEXEC_NO_UNIQUE_ADDRESS Fun fun_;
+
+            template <class Self, class Env>
+            using completion_signatures = //
+                stdexec::make_completion_signatures<
+                stdexec::__copy_cvref_t<Self, Sender>,
+                Env,
+                stdexec::completion_signatures<stdexec::set_error_t(cudaError_t)>,
+                set_value_t >;
+
+            template <stdexec::__decays_to<__t> Self, stdexec::receiver Receiver>
+                requires stdexec::
+                receiver_of<Receiver, completion_signatures<Self, stdexec::env_of_t<Receiver>>>
+                friend auto
+                tag_invoke(stdexec::connect_t, Self&& self, Receiver&& rcvr) -> stream_op_state_t<
+                stdexec::__copy_cvref_t<Self, Sender>,
+                receiver_t<Receiver>,
+                Receiver> {
+                return stream_op_state<stdexec::__copy_cvref_t<Self, Sender>>(
+                ((Self&&) self).sndr_,
+                (Receiver&&) rcvr,
```

Comment:
Something seems to be wrong with the formatting.

Reply:
Thanks a lot, I had it disabled but reenabled it :+1: